Fix `Source files` in localization document. (#25795)
kubernetes/website: content/en/docs/contribute/localization.md

### Source files

> Localizations must be based on the English files from your target version.

Reviewer: ```suggestion
Localizations must be based on the English files from a specific release targeted by the localization team.
Each localization team can decide which release to target which is referred to as the _target version_ below.
```

Localizations must be based on the English files from your target version.

To find source files for the most recent release:
To find source files for your target version:

1. Navigate to the Kubernetes website repository at https://github.com/kubernetes/website.
2. Select the `release-1.X` branch for the most recent version.
2. Select branch for your target version from followings.

> The latest version is {{< latest-version >}}, so the most recent release branch is [`{{< release-branch >}}`](https://github.com/kubernetes/website/tree/{{< release-branch >}}).

Reviewer: We wil want to keep this.
If `release-1.20` is not there yet, let's create it.

Reviewer 2: Hello @tengqm :) 
As you know, `master` branch includes the latest documents of the latest K8s release. (currently, v1.20)
I think we are using `release-x.xx` branch for maintaining k8s.io documents for previous K8s version.

So, I am not sure representing `the most recent release branch is release-x.xx` is appropriate.

 if we create `release-1.20` branch, we might need to keep updating that branch as well according to the changes in the `master` branch. 
I think release team will create `release-1.20` branch just before the K8s v1.21 release.

In case of Korea 10n team, our translation target is `master` branch (currently targeted K8s release version is v1,20),

Reviewer: Valid question. If I'm understanding this correctly, we are always treating the `master` branch as the least release. So, for release `1.20`, the `release-1.20` branch is meaningless.
If this understanding is correct, we may want to explicitly spell it out as a replacement to line 215. WDYT?

Author: Thank you for reviewing. Please check recent commit.

### Source files

Localizations must be based on the English files from a specific release targeted by the localization team.
Each localization team can decide which release to target which is referred to as the _target version_ below.

To find source files for your target version:

1. Navigate to the Kubernetes website repository at https://github.com/kubernetes/website.

> 2. Select branch for your target version from followings.

Reviewer 3: nit, possibly reword: Select a branch for your target version from the following table:

Author: done

1. Navigate to the Kubernetes website repository at https://github.com/kubernetes/website.
2. Select the `release-1.X` branch for the most recent version.
2. Select branch for your target version from followings.

> The latest version is {{< latest-version >}}, so the most recent release branch is [`{{< release-branch >}}`](https://github.com/kubernetes/website/tree/{{< release-branch >}}).

Reviewer 4: :information_source:  One thing I considered: setting up a GitHub Action to _mirror_ master into the release branch for the current release, and also blocking merges into that branch whilst it remains the mirror of the current release.

Author: I filed about reconsidering branch strategy into as issue #25806. Let's discuss at there.

Author: The discuss might take time. I'd like to describe about current workflow in this PR.

Previous version | `release-*.**`

> `master` branch holds content for the current release `{{< latest-version >}}`. And release team will create `{{< release-branch >}}` branch just before the next release, v{{< skew nextMinorVersion >}}.

Reviewer 4: ```suggestion
The `master` branch holds content for the current release `{{< latest-version >}}`. The release team will create `{{< release-branch >}}` branch shortly before the next release: v{{< skew nextMinorVersion >}}.
```

Maybe link to https://github.com/kubernetes/sig-release/tree/master/release-team/#kubernetes-release-team-roles ?

> The `master` branch holds content for the current release `{{< latest-version >}}`. The release team will create `{{< release-branch >}}` branch shortly before the next release: v{{< skew nextMinorVersion >}}. If new release branch `{{< release-branch >}}` is created before merging your localization branch to the `master` branch, switch  upstream branch for your localization branch to `{{< release-branch >}}`.

Reviewer 2: I think `switch upstream branch for your localization branch to release-1.20` is not a complete guide for the case of `if new release branch release-1.20 is created before merging your localization branch to the master branch`.

I think we need to merge the localization branch into both `master` branch and new release branch to make both branches include localized contents of the localization branch.

Since [Line:279](https://github.com/kubernetes/website/pull/25795/files#diff-6f6c393e27c210c7c735bfaf19496920276feb95fc15a977ae359fb25e76f77aR276) also provides a similar note, how about remove the sentence from this section.

```suggestion
The `master` branch holds content for the current release `{{< latest-version >}}`. The release team will create `{{< release-branch >}}` branch shortly before the next release: v{{< skew nextMinorVersion >}}.
```

Author: 👍 Thanks!

## Branching strategy

Because localization projects are highly collaborative efforts, we encourage teams to work in shared localization branches.

> To collaborate on a localization branch:

Reviewer 3: Hi @shu-mutou .
Is the main focus of this PR to change the branch terminology (development -> localization) in this guide?
I think it would be worthwhile to get some localization teams to review the changes.
@kubernetes/sig-docs-fr-reviews @kubernetes/sig-docs-ja-reviews @kubernetes/sig-docs-ko-reviews @kubernetes/sig-docs-zh-reviews 

Are you able to rebase your changes into a single commit? Thanks!

Author: > Is the main focus of this PR to change the branch terminology (development -> localization) in this guide?

No, the main purpose is the clarifications how l10n maintainers and translators should operate branches.

    Your team approvers joined the `@kubernetes/website-maintainers` team when you [added your localization team](#add-your-localization-team-in-github) to the [`kubernetes/org`](https://github.com/kubernetes/org) repository.

    We recommend the following branch naming scheme:

    `dev-<source version>-<language code>.<team milestone>`

    For example, an approver on a German localization team opens the localization branch `dev-1.12-de.1` directly against the k/website repository, based on the source branch for Kubernetes v1.12.

2. Individual contributors open feature branches based on the localization branch.

    For example, a German contributor opens a pull request with changes to `kubernetes:dev-1.12-de.1` from `username:local-branch-name`.

3. Approvers review and merge feature branches into the localization branch.

4. Periodically, an approver merges the localization branch to its source branch by opening and approving a new pull request. Be sure to squash the commits before approving the pull request.

Repeat steps 1-4 as needed until the localization is complete. For example, subsequent German localization branches would be: `dev-1.12-de.2`, `dev-1.12-de.3`, etc.

> Teams must merge localized content into the same release branch from which the content was sourced. For example, a localization branch sourced from `release-1.19` must be merged into `release-1.19`.

Reviewer 2: ```suggestion
Teams must merge localized content into the same branch from which the content was sourced. 
For example,  
  - a localization branch sourced from `master` branch must be merged into `master` branch.
  - a localization branch sourced from `release-1.19` branch must be merged into `release-1.19` branch.
```

I think we don't need to limit the source branch whether it is dev, current (master), or release branch.

Author: done

{{< note >}}

> If your localization branch was created from `master` branch but it is not merged into `master` before new release branch `{{< release-branch >}}` created, merge it into new release branch `{{< release-branch >}}`.

Reviewer 2: ```suggestion
If your localization branch was created from `master` branch but it is not merged into `master` before new release branch `{{< release-branch >}}` created, merge it into both `master` and new release branch `{{< release-branch >}}`. To merge your localization branch into new release branch `{{< release-branch >}}`, you need to switch upstream branch of your localization branch to `{{< release-branch >}}`.
```
Related with [the comment in L:222](https://github.com/kubernetes/website/pull/25795/files#r556320090)

{{< /note >}}

At the beginning of every team milestone, it's helpful to open an issue comparing upstream changes between the previous localization branch and the current localization branch. There are two scripts for comparing upstream changes. [`upstream_changes.py`](https://github.com/kubernetes/website/tree/master/scripts#upstream_changespy) is useful for checking the changes made to a specific file. And [`diff_l10n_branches.py`](https://github.com/kubernetes/website/tree/master/scripts#diff_l10n_branchespy) is useful for creating a list of outdated files for a specific localization branch.

> At the beginning of every team milestone, it's helpful to open an issue [comparing upstream changes](https://github.com/kubernetes/website/blob/master/scripts/upstream_changes.py) between the previous localization branch and the current localization branch.

Reviewer 2: ```suggestion
```
L281 seems duplicated with L279. :)

Author: done

Repeat steps 1-4 as needed until the localization is complete. For example, subsequent German development branches would be: `dev-1.12-de.2`, `dev-1.12-de.3`, etc.

> Teams must merge localized content into the same release branch from which the content was sourced. For example, a development branch sourced from `{{< release-branch >}}` must be merged into `{{< release-branch >}}`.

Reviewer 4: ```suggestion
{{< note >}}
Localization work should target the current release. If a minor (or major) Kubernetes release has happened since a localization branch was started, create a new development branch based on either `master` or on an ancestor of `master` _that includes updates for the current release_, and merge in any existing development branch that was working on documenting a previous release.
{{< /note >}}
```
seems more accurate / appropriate.

Author: Isn't it "a descendant", not "an ancestor"?
I'm drawing relationship between branches as follow and I'd like to add it into here, please take a look also.
https://hackmd.io/smrEVJCJRee3cvXTIQ7bJA

Reviewer 4: Ancestor is correct; I'm describing the **merge base** of the new localization branch that you're creating.. If after you start, someone adds a commit to _master_ and not to your branch, that new commit to _master_ does not invalidate your work so far (although you might need to address conflicts).

Author: Some l10n teams working on previous releases, so `Localization work should target the current release.` is too strict, I think.
Also, could you simplify the long sentence in note? Long sentence tends to be difficult to understand. Thanks.

Author: I rephrased this section as in new commit. PTAL.
